File: tests/CaseEndComment.cs
using System;

namespace Demo {
    public partial class CaseEndComment {
        public void Test(string val) {
            switch (val)
            {
                case 'A': Console.WriteLine("a"); break;
                //    'B':
                //      Console.WriteLine('b');
            }
        }
    }
}

File: tests/CaseEndComment.pas
namespace Demo;

uses System;

type
  CaseEndComment = public class
  public
    method Test(val: String);
  end;

implementation

method CaseEndComment.Test(val: String);
begin
  case val of
    'A': Console.WriteLine('a');
//    'B':
//      Console.WriteLine('b');
  end;
end;

end.
